docs/core_docs/docs/tutorials/retrievers.ipynb

```diff
@@ -0,0 +1,709 @@
+    "# Build a semantic search engine\n",
+    "\n",
+    "This tutorial will familiarize you with LangChain's [document loader](/docs/concepts/document_loaders), [embedding](/docs/concepts/embedding_models), and [vector store](/docs/concepts/vectorstores) abstractions. These abstractions are designed to support retrieval of data--  from (vector) databases and other sources--  for integration with LLM workflows. They are important for applications that fetch data to be reasoned over as part of model inference, as in the case of retrieval-augmented generation, or [RAG](/docs/concepts/rag) (see our RAG tutorial [here](/docs/tutorials/rag)).\n",
+    "\n",
+    "Here we will build a search engine over a PDF document. This will allow us to retrieve passages in the PDF that are similar to an input query.\n",
+    "\n",
+    "## Concepts\n",
+    "\n",
+    "This guide focuses on retrieval of text data. We will cover the following concepts:\n",
+    "\n",
+    "- Documents and document loaders;\n",
+    "- Text splitters;\n",
+    "- Embeddings;\n",
+    "- Vector stores and retrievers.\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "### Jupyter Notebook\n",
+    "\n",
+    "This and other tutorials are perhaps most conveniently run in a Jupyter notebook. See [here](https://jupyter.org/install) for instructions on how to install.\n",
+    "\n",
+    "### Installation\n",
+    "\n",
+    "This guide requires `@langchain/community` and `pdf-parse`:\n",
+    "\n",
+    "```{=mdx}\n",
+    "import Npm2Yarn from '@theme/Npm2Yarn';\n",
+    "import TabItem from '@theme/TabItem';\n",
+    "import CodeBlock from \"@theme/CodeBlock\";\n",
+    "\n",
+    "<Npm2Yarn>\n",
+    "  @langchain/community pdf-parse\n",
+    "</Npm2Yarn>\n",
+    "```\n",
+    "\n",
+    "For more details, see our [Installation guide](/docs/how_to/installation/).\n",
+    "\n",
+    "### LangSmith\n",
+    "\n",
+    "Many of the applications you build with LangChain will contain multiple steps with multiple invocations of LLM calls.\n",
+    "As these applications get more and more complex, it becomes crucial to be able to inspect what exactly is going on inside your chain or agent.\n",
+    "The best way to do this is with [LangSmith](https://smith.langchain.com).\n",
+    "\n",
+    "After you sign up at the link above, make sure to set your environment variables to start logging traces:\n",
+    "\n",
+    "```shell\n",
+    "export LANGCHAIN_TRACING_V2=\"true\"\n",
+    "export LANGCHAIN_API_KEY=\"...\"\n",
+    "\n",
+    "# Reduce tracing latency if you are not in a serverless environment\n",
+    "# export LANGCHAIN_CALLBACKS_BACKGROUND=true\n",
+    "```\n",
+    "\n",
+    "\n",
+    "## Documents and Document Loaders\n",
+    "\n",
+    "LangChain implements a [Document](https://v03.api.js.langchain.com/classes/_langchain_core.documents.Document.html) abstraction, which is intended to represent a unit of text and associated metadata. It has three attributes:\n",
```

Comment:
nit: drop the `v03`, just `https://api.js.langchain.com/classes/_langchain_core.documents.Document.html`